core/io/resource_loader.h

```diff
@@ -190,17 +188,26 @@ class ResourceLoader {
 		ResourceFormatLoader::CacheMode cache_mode = ResourceFormatLoader::CACHE_MODE_REUSE;
 		Error error = OK;
 		Ref<Resource> resource;
-		bool use_sub_threads = false;
 		HashSet<String> sub_tasks;
 
+		bool awaited : 1; // If it's in the pool, this helps not awaiting from more than one dependent thread.
+		bool need_wait : 1;
+		bool in_progress_check : 1;
```

Comment:
I'd add a comment like `// Measure against recursion cycles in progress reporting. Cycles are not expected, but can happen due to how it's currently implemented.`